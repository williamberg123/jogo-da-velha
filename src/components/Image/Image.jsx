import React from 'react';

import PropTypes from 'prop-types';

export default function Image({ imageRoot }){
    return (
        <img src={imageRoot} alt="img" />
    );
}

Image.propTypes = {
    imageRoot: PropTypes.string.isRequired
};
