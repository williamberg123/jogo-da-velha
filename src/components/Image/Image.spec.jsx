import React from 'react';
import { screen, render } from '@testing-library/react';
import Image from './Image';

describe('<Image />', () => {
    it('should render Image correctly', () => {
        render(<Image imageRoot="./img/img.png" />);

        expect(screen.getByRole('img', { name: /img/i })).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        render(<Image imageRoot="./img/img.png" />);

        expect(screen.getByRole('img', { name: /img/i })).toMatchSnapshot();
    });
});
