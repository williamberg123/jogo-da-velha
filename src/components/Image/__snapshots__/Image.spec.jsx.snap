// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should match snapshot 1`] = `
<img
  alt="img"
  src="./img/img.png"
/>
`;
